### ISSUE TITLE
Decoding data with missing keys fills in with a struct

### ISSUE THREAD

**Issue by AlejandroHuerta** [devinus/poison#102]
I'm seeing consistent behavior on 2.2.0 (came with Phoenix) which I believe is incorrect.

I have a struct which nest a few other structs:
```
defmodule PlexServer.Request do
  defstruct [:action, :x, :y, :turns, infrastructure: nil, to: nil, from: nil, ships: nil]
end
```
I receive some data such as:
```
%{"action" => :scan, "turns" => 7, "x" => 1, "y" => 1}
````

I pass this data into Poison with:
```
request = request
        |> Poison.Decode.decode(as: %PlexServer.Request{
            infrastructure: %PlexServer.Infrastructure{},
            to: %PlexServer.Tile{},
            from: %PlexServer.Tile{},
            ships: [%PlexServer.Ship{}]
        })
        |> Map.put(:at, Timex.now)
```

And what I end up getting out is:
```
%{action: :scan, at: #<DateTime(2016-10-07T21:40:48.334864Z Etc/UTC)>,
  from: %{x: nil, y: nil},
  infrastructure: %{agriculture: 0, commercial: 0, housing: 0, industry: 0,
    mining: 0}, ships: [%{count: nil, template_id: nil}], to: %{x: nil, y: nil},
  turns: 7, x: 1, y: 1}
```

What I would expect to see is the keys that were not populated would default to nil. Am I missing something or is this a bug?

**Comment by AlejandroHuerta** [comment]
I overrode the poison dependency to 3.0.0 and this behavior doesn't seem to be present anymore.

**Comment by devinus** [comment]
@AlejandroHuerta That's correct. That was specifically fixed in 3.0.
